src/test/scala/com/adazza/example/spark/job/TestWordCount.scala

```diff
@@ -0,0 +1,52 @@
+package com.adazza.example.spark.job
+
+import java.io.File
+
+import org.scalatest.FlatSpec
+
+import scala.concurrent.{ExecutionContext, Future}
+import java.util.UUID
+
+import org.junit.runner.RunWith
+import org.scalatest.junit.JUnitRunner
+
+import org.scalatest.concurrent.Eventually._
+import org.scalatest.time.SpanSugar._
+
+/**
+  * Example Spark Job unit test that utilizes the SparkJobTestHarness
+  */
+@RunWith(classOf[JUnitRunner])
+class TestWordCount extends FlatSpec with SparkJobTestHarness {
+
+  // Helper method that created an 8-character string from a random UUID
+  private def uuid8: String = UUID.randomUUID().toString.take(8)
+
+  // Helper method that takes a directory and returns a list of it's files
+  private def getListOfFiles(dir: String): List[File] = {
+    val d = new File(dir)
+    if (d.exists && d.isDirectory) {
+      d.listFiles.filter(_.isFile).toList
+    } else {
+      List.empty[File]
+    }
+  }
+
+  "WordCount" should "write output file to correct location" in {
+    val inputAbsPath = getClass.getResource("/test-files/usdeclar.txt").getPath
+    val outputAbsPath = s"/tmp/spark-test-harness-$uuid8"
```

Comment:
Don't most of these unit test frameworks already create a scratch area for capturing written files? If we use that, then presumably a "gradle clean" will remove any old cruft that got dropped.